### PR TITLE
Add more examples of with() to the cookbook

### PR DESCRIPTION
**Pull request by ptomato**
Use cases from #561 suggested by @justingrant

Closes: #561
See also: #240

### CODE REVIEW

File: docs/cookbook/adjustDayOfMonth.mjs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+// Third day of next month:
+
+const thirdOfNextMonth = date.plus({ months: 1 }).with({ day: 3 });
+
+assert.equal(thirdOfNextMonth.toString(), '2020-05-03');
+
+// Last day of this month:
+
+const lastOfThisMonth = date.with({ day: 31 });
```
Comment (sffc) [inline docs/cookbook/adjustDayOfMonth.mjs:11]: Does `date.with({ day: Infinity })` work?  (Months could in theory have more than 31 days)

Another option:

```javascript
const lastOfThisMonth = date.plus({ months: 1 }).minus({ days: 1 });
```

Comment (ptomato) [reply docs/cookbook/adjustDayOfMonth.mjs:11]: > Does `date.with({ day: Infinity })` work?

Yes, that does work in the polyfill currently, although I haven't checked the spec to see if it's intended to work.

> ```js
> const lastOfThisMonth = date.plus({ months: 1 }).minus({ days: 1 });
> ```

For an input of May 15, that will give June 14. Maybe this?

```js
date.plus({ months: 1 }).with({ day: 1 }).minus({ days: 1 })
```

Comment (sffc) [reply docs/cookbook/adjustDayOfMonth.mjs:11]: Sounds good.  Use either form, which ever you like more, or put them both into the cookbook

Comment (ptomato) [reply docs/cookbook/adjustDayOfMonth.mjs:11]: I forgot about `day: date.daysInMonth` which would be more readable than either of those two. I'll go with that.

**Pull request merged**